scripts/release.sh

```diff
@@ -119,6 +119,21 @@ main() {
     exit 1
   fi
 
+  if [ "${NO_GH_RELEASE}" == 1 ]; then
+    log_callout "Skipping gh verification, --no-gh-release is set"
```

Comment:
```suggestion
  if [ "${DRY_RUN}" == "true" ] || [ "${NO_GH_RELEASE}" == 1 ];
    log_callout "Skipping gh verification, --no-gh-release or DRY_RUN=true is set"
```

Reply:
Hi @ahrtr, this is intentional, as suggested by @jmhbnz (ref: https://github.com/etcd-io/etcd/pull/18649#discussion_r1779470541). I initially had it the way you suggested, but I agree with James that verifying the auth status would be reasonable when doing it in DRY mode.

Reply:
OK, thx for the clarification.